### ISSUE TITLE
fxpay "XHR error event" error

### ISSUE THREAD

**Issue by rob-baker** [mozilla/fxpay#112]
I have added FireFox Marketplace in-app payments to my game pathuku using fxPay.

Everything seems to be working when i use the fakeProducts setting below:

    fxpay.configure({fakeProducts: true});

When i try to use fxPay in Live i get the following in my console:

    ok
    23:30:46.305 "opening" "GET" "to" "https://marketplace.firefox.com/api/v1/payments/https%3A%2F%2Fwww.pathuku.com/in-app/?active=1"1 fxpay.min.js:1:476
    
    error
    23:30:46.681 "XHR error event:" error { target: XMLHttpRequest, isTrusted: true, lengthComputable: false, loaded: 0, total: 0, currentTarget: XMLHttpRequest, eventPhase: 2, bubbles: false, cancelable: false, defaultPrevented: false, timeStamp: 1432593046679000 }1 fxpay.min.js:1:4078
    
    error
    23:30:46.690 "Error getting products:ApiRequestError: received XHR error for path: /payments/https%3A%2F%2Fwww.pathuku.com/in-app/?active=1"1 main.js:1:10765
     _eo/b.e5.fxpay.checkForPurcs/<() main.js:1
    z() fxpay.min.js:1
    A() fxpay.min.js:1
    x() fxpay.min.js:1
    t() fxpay.min.js:1
    j() fxpay.min.js:1

A http OPTIONS request is made to 
    https://marketplace.firefox.com/api/v1/payments/https%3A%2F%2Fwww.pathuku.com/in-app/?active=1

The url seems to be working, but i don't understand why only a OPTIONS request is being made?

My game urls are
https://marketplace.firefox.com/app/pathuku/
https://www.pathuku.com/manifest.webapp
https://www.pathuku.com

Do i need a "type" in my manifest? Ideally i want the in-app payments to work from any platform not just FireFox OS.

Any help would be very welcome

**Comment by kumar303** [comment]
Hi. I'm not sure what's going on at a glance. Here are some questions/answers:

* what version of fxpay are you using? Did you try it using [the latest](https://github.com/mozilla/fxpay/releases)?
* the OPTIONS request is made by the platform to make sure CORS is working. It looks like it is working because you received a GET later and also because fake products are working.
* Your app looks like it's approved and public so that's good. Are all your products configured in the DevHub and marked active?
* You don't need to adjust your type in the manifest because your app is hosted (I already see ``type:web`` which is fine).
* Could you paste in an entire console log? You can do so by running your app in the WebIDE and [opening the console](https://developer.mozilla.org/en-US/docs/Tools/WebIDE/Running_and_debugging_apps#Debugging_apps). Then select-all on the log panel, copy, and paste.

**Comment by rob-baker** [comment]
I found the issue after lots of debugging.

I was loading the old json.js file, this adds toJSONString and parseJSON methods to the Object.prototype.

When i called fxpay.getProducts() i guess some objects get serialised and the methods were caused issues.

I have removed the reference to json.js as i believe this has now been added natively to most browsers , everything is working now.

Thanks you for your help. I didn't know how powerful the WebIDE was.
